cmake/tribits/devtools_install/0001-CTest-Add-CTEST_RESOURCE_SPEC_FILE-variable.patch

```diff
@@ -0,0 +1,377 @@
+new file mode 100644
+index 0000000..321e9a2
+--- /dev/null
++++ b/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-check.cmake
+@@ -0,0 +1,3 @@
++if(EXISTS "${RunCMake_TEST_BINARY_DIR}/ctresalloc.log")
++  set(RunCMake_TEST_FAILED "ctresalloc.log should not exist")
++endif()
+diff --git a/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-result.txt b/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-result.txt
+new file mode 100644
+index 0000000..b57e2de
+--- /dev/null
++++ b/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-result.txt
+@@ -0,0 +1 @@
++(-1|255)
+diff --git a/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-stderr.txt b/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-stderr.txt
+new file mode 100644
+index 0000000..dcf13e0
+--- /dev/null
++++ b/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable-stderr.txt
+@@ -0,0 +1,14 @@
++^Insufficient resources for test Test1:
++
++  Test requested resources of type 'fluxcapacitors' in the following amounts:
++    200 slots
++  but only the following units were available:
++    'outatime': 121 slots
++
++Resource spec file:
++
++  [^
++]*/Tests/RunCMake/CTestResourceAllocation/resspec.json
++CMake Error at [^
++]*/Tests/RunCMake/CTestResourceAllocation/notenough1-ctest-s-res-variable/test\.cmake:[0-9]+ \(message\):
++  Tests did not pass$
+diff --git a/Tests/RunCMake/CTestResourceAllocation/test.cmake.in b/Tests/RunCMake/CTestResourceAllocation/test.cmake.in
+index 4b426f1..54eb4e9 100644
+--- a/Tests/RunCMake/CTestResourceAllocation/test.cmake.in
++++ b/Tests/RunCMake/CTestResourceAllocation/test.cmake.in
+@@ -8,15 +8,21 @@ set(CTEST_CMAKE_GENERATOR_TOOLSET "@RunCMake_GENERATOR_TOOLSET@")
+ set(CTEST_BUILD_CONFIGURATION "$ENV{CMAKE_CONFIG_TYPE}")
+ set(CTEST_NIGHTLY_START_TIME "01:00:00 UTC")
+ 
+-ctest_start(Experimental QUIET)
+-ctest_configure(OPTIONS
++set(config_options
+   "-DCTEST_RESOURCE_ALLOC_ENABLED=${CTEST_RESOURCE_ALLOC_ENABLED};-DCTRESALLOC_COMMAND=${CTRESALLOC_COMMAND}"
+   )
+-ctest_build()
+ 
+-if(CTEST_RESOURCE_ALLOC_ENABLED)
++if(CTEST_RESOURCE_SPEC_SOURCE STREQUAL "ARG")
+   set(resspec RESOURCE_SPEC_FILE "@RunCMake_SOURCE_DIR@/resspec.json")
++elseif(CTEST_RESOURCE_SPEC_SOURCE STREQUAL "VARIABLE")
++  set(CTEST_RESOURCE_SPEC_FILE "@RunCMake_SOURCE_DIR@/resspec.json")
++elseif(CTEST_RESOURCE_SPEC_SOURCE STREQUAL "CACHE")
++  list(APPEND config_options "-DCTEST_RESOURCE_SPEC_FILE=@RunCMake_SOURCE_DIR@/resspec.json")
+ endif()
++
++ctest_start(Experimental QUIET)
++ctest_configure(OPTIONS "${config_options}")
++ctest_build()
+ ctest_test(${resspec} RETURN_VALUE retval PARALLEL_LEVEL ${CTEST_PARALLEL} SCHEDULE_RANDOM ${CTEST_RANDOM})
+ if(retval)
+   message(FATAL_ERROR "Tests did not pass")
+-- 
+2.7.4
+
```

Comment:
Is this `*.patch` file supposed to be in this PR?

Reply:
Updates from TriBITS are just snapshotted for TriBITS 'master'.  It would be a mess to try to only snapshot in individual TriBITS branches.